src/services/message.service.ts

```diff
@@ -75,6 +126,38 @@ export class MessageService {
     return msg;
   }
 
+  private _constructMessages(
+    splitMessage: string[],
+    templateLength: number,
+    delimiter: string,
+    header: string,
+    footer: string
+  ): string[] {
+    const messagesToSend = [];
+    const numOfFreeChars = Constants.MaxMessageLength - templateLength;
+
+    let temp = '';
+
+    while (splitMessage.length > 0) {
+      let nextCapsule = splitMessage.shift() || '';
```

Comment:
`shift` is really inefficient, just pop, and reverse at the end. [Ref](https://github.com/cs-discord-at-ucf/lion/pull/336#discussion_r605117249)

Reply:
The addition of the title option makes the code less readable and more repetitive so I don't think it is worth the performance gain.

Reply:
What does this have to do with the title?
https://github.com/CallumCoy/lion/blob/b466497018653d926536b11f4aced512f05759af/src/services/message.service.ts#L58
reverse this
https://github.com/CallumCoy/lion/blob/b466497018653d926536b11f4aced512f05759af/src/services/message.service.ts#L142-L148
And pop instead of shift

Reply:
I know what you are telling me to do, but as I said it makes the code less readable since you end up adding the title after you split the message.  It also makes the code repetitive since you exit the while loops, but then have to enter a similar one to split apart the title.  Which as I said, I don't think adding that complexity is worth the potential performance boost.  And the other option is to make the while loops have branches lowering the readability of those while loops that are already not very readable.

Reply:
For reference, it would look something like [this](https://www.codepile.net/pile/kBlmM75r)  current code at line 70.